src/views/Home.jsx

```diff
@@ -1,11 +1,29 @@
+import { useEffect } from 'react';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { useNavigate } from 'react-router-dom';
 import './Home.css';
 
-export function Home() {
+export function Home({ handleListTokenState, listToken }) {
+	const navigate = useNavigate();
+
+	const handleCreateList = () => {
+		const newToken = generateToken();
+		handleListTokenState(newToken);
+		navigate('/list');
+	};
+
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+		}
+	}, []);
+
```

Comment:
Don't forget to add `listToken` as a dependency here.
Also, does `listToken` need to be passed as a prop for this component?

Reply:
Thanks @mentalcaries Ah, so right now we're doing just an initial render [] as the dependency because we have the `navigate('/list') in the `handleCreateList` function it's working as it should.

So if I add `listToken` as a dependency in the `useEffect` I can remove the `navigate` from the `handleCreateList` function because the `useEffect` will trigger once it's added!  Thank you! That will remove an unnecessary navigate from the function and it seems to be having the same effect.

Side note -- is it strange that when I delete the token from my local storage that I have to refresh for it to work and show`Home` again?

To answer your question about `listToken` as a prop, I think we do need to pass it down because this `listToken` state is being set in App.jsx and we need it to be able to use it in useNavigate.  Otherwise we'd have to create a new state in Home.jsx which we could do, but I think that might be adding something unnecessary.  Unless I'm missing something?

Reply:
So here's one idea(which may or may not be valid): If the `listToken` is being set in `App.jsx`, can the navigation to `/list` be done there (presumably int he `useEffect` there,  instead of having to pass it down to another component?

Reply:
@mentalcaries so the reason we didn't navigate in `App.jsx` is unfortunately the `useNavigate` won't work at the `App.jsx` component level because the Routes are just being set up there.  So basically you can't declare `useNavigate` in the same place you're setting up `BrowserRouter`.  We could theoretically move `BrowserRouter` up a level and wrap `index.jsx` in `BrowserRouter` to get that to work maybe?  But I think for the purposes of this we probably shouldn't start messing with `index.jsx`?

Reply:
That's a good point! You're right about having to move `BrowserRouter` up a level.  I agree, it's probably better to take a simpler approach in this case. 

Just for the sake of bouncing ideas around, what do you think about using the presence of the `listToken` to conditionally render components in `App.jsx`? Also React router has a [`<Navigate />` component](https://reactrouter.com/en/main/components/navigate) that you can look at `if` you fancy React Router.

Reply:
@mentalcaries absolutely to both ideas! I know when @Yaosaur and I were bouncing ideas around she had noted an option of utilizing the `Navigate` component option.  Do you know in general if there is a preferred or best practice way to do it because I'm down for either.

Reply:
I'm not sure if there's a best practice, but I do think the `Navigate` component would allow the redirects without having to move the Router tags up a level